pkg/network/namescheme/networknamescheme.go

```diff
@@ -41,19 +41,42 @@ const (
 // The returned map associates between the network name and the generated pod interface name.
 // Primary network will use "eth0" and the secondary ones will be named with the hashed the network name (sha256).
 func CreateNetworkNameScheme(vmiNetworks []v1.Network) map[string]string {
-	networkNameSchemeMap := mapMultusNonDefaultNetworksToPodInterfaceName(vmiNetworks)
+	return createNameSchemeByKind(vmiNetworks, hashedNameScheme)
+}
+
+// CreateOrdinalNetworkNameScheme iterates over the VMI's Networks, and creates for each a pod interface name.
+// The returned map associates between the network name and the generated pod interface name.
+// Primary network will use "eth0" and the secondary ones will use "net<id>" format, where id is an enumeration
+// from 1 to n.
+func CreateOrdinalNetworkNameScheme(vmiNetworks []v1.Network) map[string]string {
```

Comment:
And at the same time the name is also about abstracting away the "how".
At the caller of this function, I want to know it handles legacy stuff and if I delve in (in case I care), it will be nice to read it does it using ordinal names.

I do not think it is clear at the caller that this handles legacy VMI/s.
If it is, I take it back.

Reply:
I still think the whole thing about these 2 functions is focused on the "how" the names are generated. 

Let me give you an example: 5 years from now we (for whatever reason) have to add another naming scheme. One is called legacy, the "non-legacy" one would need to be renamed, and the new one would be called ?... NewOne? Not fond of that. 

> I do not think it is clear at the caller that this handles legacy VMI/s.

And what's a legacy VM ? ... Is that going to be obvious for everyone ? And furthermore, can we say / promise that "legacy VM" will mean **one thing** across different domains ?